pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -266,6 +266,13 @@ func (m *kubeGenericRuntimeManager) makeMounts(opts *kubecontainer.RunContainerO
 
 	for idx := range opts.Mounts {
 		v := opts.Mounts[idx]
+		if _, err := m.osInterface.Stat(v.HostPath); os.IsNotExist(err) {
```

Comment:
xref #46597.

This is a breaking change for volume plugin `HostPath`, although no harm comes. Only when the type of `hostpath` in the `PodSpec` is set to `"DirectoryOrCreate"`, a new dir will be created automatically. The default behavior is a no-op, which is mainly for backward compatibility. In this change, when the `HostPathType` is unset, a new dir will still be created, this is not what we suggested. 

For other types of volume plugins, I think we should let the plugin mounter handle it independently as well. WDYT?

Reply:
> The default behavior is a no-op, which is mainly for backward compatibility. In this change, when the HostPathType is unset, a new dir will still be created, this is not what we suggested.

Although there is no-op in volume manager, docker will create a new dir for the hostPath. This change won't change this behavior and aligns all other CRI runtimes with the same behavior.

Reply:
> docker will create a new dir for the hostPath. This PR won't change this behavior and aligns all other CRI runtimes with the same behavior.

AFAIK Rkt won't create such dir if not existed.

moby/moby#21666 added back the deprecated auto-creation of host-directories, although no consensus had been reached.

Just as the hostpath type proposal doc #34058 said, 

> This is rarely what the user actually wants because hostPath volumes are typically used to express a dependency on an existing external file or directory.

> it was suggested that orchestration systems could better manage volume creation than Docker, but Docker does so as well anyways.

IMO it is kubelet that should be responsible for such auto-creation, instead of docker, rkt, or other CRI runtimes.

Reply:
> This is what this PR trying to do. I think we should define the logic clearly in kubelet and CRI runtimes shouldn't care about this issue.

Let the volume plugins/mgrs handle it. WDYT?

And this PR is breaking the behavior of volume plugin `HostPath`. *For non-existent dir, a new dir should not be created unless `HostPathType` is set to `DirectoryOrCreate`.*

Reply:
> Let the volume plugins/mgrs handle it. WDYT?

LGTM. Since hostPath volume plugin has already create the path if its type if DirectoryOrCreate,  CRI is aligned with volume's behavior if it requires runtimes to report an error. Let me drop this code change.

Reply:
@dixudx Updated. PTAL

Reply:
I don't understand the discussion here. I think @dixudx is right. We should not enforce `if the hostPath doesn't exist, then runtimes should report error`, because that breaks the backward compatibility. *ALL existing pods using non-existent host path will fail. That is unacceptable.*

We should define `If the hostPath doesn't exist, runtime should create it` instead.
@feiskyer @yujuhong

Reply:
NVM, saw https://github.com/kubernetes/kubernetes/pull/46597/files#r130686032 which makes sense to me.

Given so, I think we should either not define this in CRI or define `If the hostPath doesn't exist, runtime should create a directory with 0755`.

Reply:
If so, it's a same operation for all runtimes. why not create the dir in kubelet instead?

Reply:
@feiskyer We can default `HostPathUnset` to create directory in kubelet if not exist, but we are not doing that today.

Given this is not finalized, we should at least remove current definition which doesn't work with today's kubelet implementation.

Reply:
How about adding creation logic in kuberuntime and keep CRI as is? In such way, runtimes should still error if the hostpath doesn't exist, right?